scripts/verify.sh

```diff
@@ -40,10 +40,36 @@ if [ $final_ret -eq 0 ]; then
     if [ "${diff#-}" -gt 500000 ]; then
         final_ret=4
     fi
+    # Ping test result
+    sudo ip netns add sta0
+    sudo ip netns add sink
+    # Create macvlan bridge mode interface.
+    # All stations are directly connected to each other with a simple bridge via the physical interfaces (owl0/owl0sink).
+    # After raw packet enter macvlan, it will follow normal kernel l2/l3 data path traffic process.
+    #  macvlan0 192.168.0.1/24 <---> owl0 <---> vwifi driver <---> owl0sink <---> macvlan1 192.168.0.2/24
+    sudo ip link add macvlan0 link owl0 type macvlan mode bridge
+    sudo ip link add macvlan1 link owl0sink type macvlan mode bridge
+
+    sudo ip link set macvlan0 netns sta0
+    sudo ip link set macvlan1 netns sink
+
+    sudo ip netns exec sta0 ip link set macvlan0 up
+    sudo ip netns exec sta0 ip addr add 192.168.0.1/24 dev macvlan0
+
+    sudo ip netns exec sink ip link set macvlan1 up
+    sudo ip netns exec sink ip addr add 192.168.0.2/24 dev macvlan1
+
+    sudo ip netns exec sta0 ping 192.168.0.2 -c 4
```

Comment:
We should avoid using `192.168.0.2` since it is defaulted by QEMU.
See https://www.qemu.org/docs/master/system/qemu-manpage.html

I am using [lima](https://github.com/lima-vm/lima) on macOS, and the above script prevents me from performing tests because the SSH connection was lost.

Reply:
I could change to other local IP such as 10.0.0.1/24 and 10.0.0.2/24 
Is change IP domain could help SSH connection?

Reply:
I will take some time to run on lima to test.

Reply:
Run the first owl0 interface up command:
```
dutsai@lima-default:/Users/dutsai/Workspace_mac/vwifi$ sudo ip link set owl0 up
dutsai@lima-default:/Users/dutsai/Workspace_mac/vwifi$
Message from syslogd@lima-default at Apr 25 11:46:17 ...
 kernel:[  130.166013] Internal error: Oops: 96000047 [#1] SMP
```
Kernel Mesage:
```
Apr 25 11:46:17 lima-default kernel: [  130.162209] Sta owl0 Tx 00000000: 33 33 00 00 00 16 00 6f 77 6c 30 00 86 dd 60 00
Apr 25 11:46:17 lima-default kernel: [  130.162211] Sta owl0 Tx 00000010: 00 00 00 24 00 01 00 00 00 00 00 00 00 00 00 00
Apr 25 11:46:17 lima-default kernel: [  130.162211] Sta owl0 Tx 00000020: 00 00 00 00 00 00 ff 02 00 00 00 00 00 00 00 00
Apr 25 11:46:17 lima-default kernel: [  130.162211] Sta owl0 Tx 00000030: 00 00 00 00 00 16 3a 00 05 02 00 00 01 00 8f 00
Apr 25 11:46:17 lima-default kernel: [  130.162212] Sta owl0 Tx 00000040: 3f 1e 00 00 00 01 04 00 00 00 ff 02 00 00 00 00
Apr 25 11:46:17 lima-default kernel: [  130.162212] Sta owl0 Tx 00000050: 00 00 00 00 00 01 ff 6c 30 00
Apr 25 11:46:17 lima-default kernel: [  130.162224] Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000
Apr 25 11:46:17 lima-default kernel: [  130.163529] Mem abort info:
Apr 25 11:46:17 lima-default kernel: [  130.163674]   ESR = 0x96000047
Apr 25 11:46:17 lima-default kernel: [  130.163865]   EC = 0x25: DABT (current EL), IL = 32 bits
Apr 25 11:46:17 lima-default kernel: [  130.164144]   SET = 0, FnV = 0
Apr 25 11:46:17 lima-default kernel: [  130.164303]   EA = 0, S1PTW = 0
Apr 25 11:46:17 lima-default kernel: [  130.164465] Data abort info:
Apr 25 11:46:17 lima-default kernel: [  130.164621]   ISV = 0, ISS = 0x00000047
Apr 25 11:46:17 lima-default kernel: [  130.164823]   CM = 0, WnR = 1
Apr 25 11:46:17 lima-default kernel: [  130.164993] user pgtable: 4k pages, 48-bit VAs, pgdp=0000000134de6000
Apr 25 11:46:17 lima-default kernel: [  130.165337] [0000000000000000] pgd=000000012ca39003, pud=0000000131b3c003, pmd=000000012db14003, pte=0000000000000000
Apr 25 11:46:17 lima-default kernel: [  130.166013] Internal error: Oops: 96000047 [#1] SMP
```

Reply:
Is lima could use the GDB ?

Reply:
lima is based on QEMU with HVF accelerator, and you can use QEMU along with GDB.

Reply:
```
Apr 30 06:58:53 lima-default kernel: [  185.867399] owl_ndo_start_xmit:580 np: 0xb2cf98c0
Apr 30 06:58:53 lima-default kernel: [  185.867400] owl_ndo_start_xmit:581 np->rx_queue: 0xb2cf9d80
Apr 30 06:58:53 lima-default kernel: [  185.867400] owl_ndo_start_xmit:582 dest_np: 0xb2cfd8c0
Apr 30 06:58:53 lima-default kernel: [  185.867401] owl_ndo_start_xmit:583 dest_np->rx_queue: 0xb2cfdd80
Apr 30 06:58:53 lima-default kernel: [  185.867401] owl_ndo_start_xmit:584 dest_np->rx_queue.next: 0x0
Apr 30 06:58:53 lima-default kernel: [  185.867402] owl_ndo_start_xmit:585 dest_np->rx_queue.prev: 0x0
Apr 30 06:58:53 lima-default kernel: [  185.867409] Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000
```